### ISSUE TITLE
[ROCm 5.3] OpenCL kernels compilation issues [-Werror,-Wunused-but-set-variable]

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#1732]
@atamazov 
Likely due to HIP or COMgr updates, these warnings are treated as errors now. Before these pooling kernels are replaced by CK we may want to disable the warning ?

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/a068b0e205f620f15d6a2c828d035d06854bfd3e/src/kernels/MIOpenPoolingBwd.cl#L289

```
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, HIP version 5.3.22334, MIOpen version 2.18.0.f1263d7a3
MIOpen(HIP): Info [PrintVersionImpl] COMgr v.2.4.0, USE_HIP_PCH: 1
MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-b9fdf3/input/MIOpenPoolingBwd.cl:333:9: error: variable 'b_idx' set but not used [-Werror,-Wunused-but-set-variable]
    int b_idx = bt_y * mlo_bot_width + bt_x;
        ^
1 error generated.
```

And I am not sure why it claims the variable is not used at https://github.com/ROCmSoftwarePlatform/MIOpen/blob/a068b0e205f620f15d6a2c828d035d06854bfd3e/src/kernels/MIOpenPoolingBwd.cl#L358

**Comment by junliume** [comment]
@atamazov  and @JehandadKhan
I can add a line right after:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/a068b0e205f620f15d6a2c828d035d06854bfd3e/src/kernels/MIOpenPoolingBwd.cl#L289
like this:
```
  int b_idx = bt_y * mlo_bot_width + bt_x;
  (void) b_idx;
```
and the issue goes away ... 
Maybe the compiler thinks `MLO_POOLBWD_N_VERT_OUT_PIX` will always be zero somehow?

**Comment by pfultz2** [comment]
We should only pass `-Werror` when `BUILD_DEV=On` so it only shows as an error in our CI and not in release versions. I am not sure if thats the case here.

> And I am not sure why it claims the variable is not used at 

The value in the variable is never read. Its incremented by some other value but we never read the value in the variable. Considering this code has not been changed in years, the variable can be safely removed with no change in functionality.

**Comment by atamazov** [comment]
Fixed in #1729

@pfultz2 
> We should only pass -Werror when BUILD_DEV=On

Of course, and this is how MIOpen works now.
